lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1836,6 +1837,11 @@ signTransaction tl preferredLatestEra witCountCtx keyLookup mextraRewardAcc
 type MakeRewardAccountBuilder k =
     (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
 
+data ErrWriteTxEra
+    = ErrOldEraNotSupported Cardano.AnyCardanoEra
+    | ErrTxNotInNodeEra Write.AnyRecentEra
```

Comment:
How about the following names:

```suggestion
    = ErrTxEraTooOld Cardano.AnyCardanoEra
    -- ^ The transaction is from an era that is too old,
    -- and no longer supported by 'balanceTransaction'.
    | ErrTxEraTooNew Write.AnyRecentEra
    -- ^ The transaction is from an era that is too new,
    -- from the point of view of the node.
```

The thing that bugs me about the current names is that `TxNotInNodeEra` doesn't really make the time asymmetry clear.

My interpretation: this error can only happen if the node has not yet caught up with the transaction era: in other words, the transaction is from an era that is "too new" from the point of view of the node. Is that also your interpretation?

Reply:
Would be happy to discuss options to make this clearer